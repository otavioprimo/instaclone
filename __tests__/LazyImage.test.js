/**
 * @format
 */

import 'react-native';
import React from 'react';
import LazyImage from '../src/components/LazyImage';
import server from '../server.json';


// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

it('create component', () => {
  renderer.create(<LazyImage />);
});

it('has props corretly', () => {
  const data = server.feed[0];
  const component = <LazyImage aspectRatio={1} shouldLoad={false} smallSource={data.small} source={data.image} />
  expect(component.props.aspectRatio).toBe(1);
  expect(component.props.shouldLoad).toBe(false);
  expect(component.props.smallSource).toBe(data.small);
  expect(component.props.source).toBe(data.image);
});

it('render snapshot', () => {
  const tree = renderer.create(<LazyImage />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('render when pass props', () => {
  const data = server.feed[0];
  const component = <LazyImage shouldLoad={false} aspectRatio={1} smallSource={data.small} source={data.image}/>
  
  const tree = renderer.create(component).toJSON();
  expect(tree).toMatchSnapshot();
})