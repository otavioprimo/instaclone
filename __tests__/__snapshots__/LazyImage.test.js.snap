// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render snapshot 1`] = `
<View
  accessibilityIgnoresInvertColors={true}
  style={
    Array [
      Object {
        "aspectRatio": 1,
        "width": "100%",
      },
    ]
  }
>
  <Image
    blurRadius={1.5}
    ratio={1}
    resizeMode="contain"
    source={
      Object {
        "uri": undefined,
      }
    }
    style={
      Array [
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        Object {
          "height": undefined,
          "width": undefined,
        },
        undefined,
      ]
    }
  />
</View>
`;

exports[`render when pass props 1`] = `
<View
  accessibilityIgnoresInvertColors={true}
  style={
    Array [
      Object {
        "aspectRatio": 1,
        "width": "100%",
      },
    ]
  }
>
  <Image
    blurRadius={1.5}
    ratio={1}
    resizeMode="contain"
    source={
      Object {
        "uri": "https://rocketseat-cdn.s3-sa-east-1.amazonaws.com/instagram-clone/small/1.jpeg",
      }
    }
    style={
      Array [
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        Object {
          "height": undefined,
          "width": undefined,
        },
        undefined,
      ]
    }
  />
</View>
`;
